api/src/main/java/org/openmrs/module/ugandaemrreports/definition/data/evaluator/DueForViralLoadPatientDataEvaluator.java

```diff
@@ -0,0 +1,72 @@
+package org.openmrs.module.ugandaemrreports.definition.data.evaluator;
+
+import org.joda.time.LocalDate;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.evaluator.CohortDefinitionEvaluator;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.openmrs.module.ugandaemrreports.common.StubDate;
+import org.openmrs.module.ugandaemrreports.definition.data.definition.DueForViralLoadPatientDataDefinition;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+
+@Handler(supports = {DueForViralLoadPatientDataDefinition.class})
+public class DueForViralLoadPatientDataEvaluator
+        implements CohortDefinitionEvaluator {
+    @Autowired
+    EvaluationService evaluationService;
+    @Override
+    public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext evaluationContext) throws EvaluationException {
+        EvaluatedCohort ret = new EvaluatedCohort(cohortDefinition, evaluationContext);
+
+        DueForViralLoadPatientDataDefinition def = (DueForViralLoadPatientDataDefinition) cohortDefinition;
+
+        LocalDate startDate = StubDate.dateOf(def.getStartDate());
+        LocalDate endDate = StubDate.dateOf(def.getEndDate());
+
+        String endDateString = endDate.toString("yyyy-MM-dd");
+        String startDateString = startDate.toString("yyyy-MM-dd");
+
+//        the query retrieves all clients that have a 12 months difference between there last vl date and are also above
+//                15 yrs,it also picks those that have a 6months difference from last vl date and are below 15
+
+        String query ="SELECT id FROM \n" +
```

Comment:
@balukujr When breaking the SQL I suggest you do it at appropriate points at UNION is where I recommend so that its readable